text/utilities/0000-syft-buildpack.md

```diff
@@ -0,0 +1,45 @@
+# Syft Buildpack
+
+## Summary
+
+Introduce a Syft buildpack. This buildpack will provide the `syft` binary should any buildpacks want to use it. The `syft` CLI is a tool and library for generating a Software Bill of Materials from container images and filesystems.
+
+The Syft buildpack would be maintained by the existing [Utilities team](https://github.com/orgs/paketo-buildpacks/teams/utilities). It uses libpak & libcnb.
+
+## Motivation
+
+It can be used by buildpacks to scan and generate both CycloneDX and SPDX listings of layers for compliance with [Buildpacks RFC #95](https://github.com/buildpacks/rfcs/blob/main/text/0095-sbom.md).
+
+## Detailed Explanation
+
+1. A Syft buildpack has been developed. It will be contributed to Paketo.
+2. Other buildpacks wishing to compress binaries will opt-in by doing the following:
+   1. At detect time, require `syft`.
+   2. At build time after generating a binary, run `syft packages -q -o <format> "dir:<path>"` to scan a directory. Valid formats are `json`, `text`, `table`, `cyclonedx`, `spdx-tag-value`, and `spdx-json`.
+
+## Rationale and Alternatives
+
+1. `syft` could be added to the various build images. This is an easy option, but increases the build image size. It also does not allow a buildpack to guarantee that Syft is present, as some users have customized build images.
+2. `syft` could be added into individual buildpacks, such as directly in the maven buildpack. Given that there are likely other buildpacks which can make use of Syft, this requires repetition of code across multiple buildpacks.
+
+## Implementation
+
+The new Syft buildpack will participate all the following conditions are met
+
+* Another buildpack requires `syft`
+
+When conditions are met, the buildpack will do the following:
+
+* Contributes Syft to a layer marked `build` and `cache` with command on `$PATH`
+
+Buildpacks that wish to opt-in, simply need to indiate that they require `syft` at detection time. Then `syft` should be present on the path at build time and these buildpacks can run it.
```

Comment:
Would you anticipate also adding in a version to the `provides/requires` relationship? It looks like syft is pre 1.0.0 so i could imagine there will be breaking changes coming at some point.

Reply:
That's certainly possible. Both breaking changes and having a way to version this. We typically handle this with buildpack by locking the buildpack version, since buildpacks are tightly coupled to the dependency versions they install.

In short, if something breaks we either don't update the Syft buildpack immediately, or we lock the consuming buildpacks to use the last known working version.

We will follow the Paketo RFC on Semver, so if things change upstream and that breaks buildpacks, I'd expect that to occur across major versions only. 

If you feel strongly about it we could explore adding support for multiple major version branches, so like we'd have 0.x and 1.x (when that's released). I could see other buildpacks passing the version through metadata. We're not doing that with other buildpacks like UPX or Watchexec though, so I would rather cross that bridge when we get to it, if it ends up being necessary.

Reply:
Oh, gotcha. So we would automatically bump the version of this buildpack when there's a new version of syft, and if it happens to break then language-family buildpacks (e.g. Java) lock to the last-known good version of this buildpack, rather than the specific downstream buildpacks (e.g. Maven) locking to a version of the dependency in the `provides`/`requires` relationship?

I'm definitely not looking to add more work and process for the sake of it!

Reply:
Yes, exactly.